integration-tests/load/vrfv2/vrfv2_test.go

```diff
@@ -254,65 +165,158 @@ func TestVRFV2Performance(t *testing.T) {
+			l.Info().
+				Str("Network Name", testEnv.EVMClient.GetNetworkName()).
+				Msg("Network is a simulated network. Skipping fund return for Coordinator Subscriptions.")
+		} else {
+			if *vrfv2Config.General.CancelSubsAfterTestRun {
+				//cancel subs and return funds to sub owner
+				vrfv2.CancelSubsAndReturnFunds(testcontext.Get(t), vrfContracts, eoaWalletAddress, subIDs, l)
 			}
-			fundingAtLeast := conversions.EtherToWei(big.NewFloat(*cfg.ExistingEnvConfig.NodeSendingKeyFundingMin))
-			fundingToSendWei := new(big.Int).Sub(fundingAtLeast, sendingKeyBalance)
-			fundingToSendEth := conversions.WeiToEther(fundingToSendWei)
-			if fundingToSendWei.Cmp(big.NewInt(0)) == 1 {
-				l.Info().
-					Str("Sending Key", sendingKey).
-					Str("Sending Key Current Balance", sendingKeyBalance.String()).
-					Str("Should have at least", fundingAtLeast.String()).
-					Str("Funding Amount in ETH", fundingToSendEth.String()).
-					Msg("Funding Node's Sending Key")
-				err := actions.FundAddress(client, sendingKey, fundingToSendEth)
-				if err != nil {
-					return err
-				}
-			} else {
-				l.Info().
-					Str("Sending Key", sendingKey).
-					Str("Sending Key Current Balance", sendingKeyBalance.String()).
-					Str("Should have at least", fundingAtLeast.String()).
-					Msg("Skipping Node's Sending Key funding as it has enough funds")
+		}
+		if !*vrfv2Config.General.UseExistingEnv {
+			if err := testEnv.Cleanup(); err != nil {
+				l.Error().Err(err).Msg("Error cleaning up test environment")
 			}
 		}
 	}
-	return nil
-}
 
+	newEnvConfig := vrfcommon.NewEnvConfig{
+		NodesToCreate:          []vrfcommon.VRFNodeType{vrfcommon.VRF},
+		NumberOfTxKeysToCreate: *vrfv2Config.General.NumberOfSendingKeysToCreate,
+		UseVRFOwner:            true,
+		UseTestCoordinator:     true,
+	}
+
+	testEnv, vrfContracts, vrfKey, _, err = vrfv2.SetupVRFV2Universe(testcontext.Get(t), t, testConfig, cleanupFn, newEnvConfig, l)
+	require.NoError(t, err)
+
+	var consumers []contracts.VRFv2LoadTestConsumer
+	subIDs, consumers, err = vrfv2.SetupSubsAndConsumersForExistingEnv(
+		testEnv,
+		vrfContracts.CoordinatorV2,
+		vrfContracts.LinkToken,
+		1,
+		*vrfv2Config.General.NumberOfSubToCreate,
+		testConfig,
+		l,
+	)
+	vrfContracts.VRFV2Consumers = consumers
+
+	eoaWalletAddress = testEnv.EVMClient.GetDefaultWallet().Address()
+
+	l.Debug().Int("Number of Subs", len(subIDs)).Msg("Subs involved in the test")
+	for _, subID := range subIDs {
+		subscription, err := vrfContracts.CoordinatorV2.GetSubscription(testcontext.Get(t), subID)
+		require.NoError(t, err, "error getting subscription information for subscription %d", subID)
+		vrfv2.LogSubDetails(l, subscription, subID, vrfContracts.CoordinatorV2)
+	}
+
+	t.Run("vrfv2 and bhs performance test", func(t *testing.T) {
+		configCopy := testConfig.MustCopy().(tc.TestConfig)
+		//Underfund Subscription
+		configCopy.VRFv2.General.SubscriptionFundingAmountLink = ptr.Ptr(float64(0.000000000000000001)) // 1 Juel
```

Comment:
Consider keeping constants for token denominators (especially those that are hard to read and easy to make a mistake if you miss one zero).

Reply:
changed to 0 just to make things simpler